### DISCUSSION TITLE
Elsa Studio 3.1 Re-invoking the Suspended HttpEndpoint Workflow with workflowInstanceId doesn't work

### DISCUSSION THREAD

**Discussion by 7studios** [elsa-workflows/elsa-core#5663] [General]
I have created a test workflow in Elsa Studio 3.1 with "Publish Event" and "Event" activities. I invoke the workflow "POST" without a body or params to get it into a suspended state. 

<img width="964" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/852585/72b86501-8031-4298-ae9e-5d40516080dc">

<img width="954" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/852585/16785171-9736-4c76-98f7-820f0f8de591">

Post the following in Postman -

{
  "workflowInstanceId": "883723cf6ae139df",
  "eventName": "Approve",
  "data": {
    "approverId": 1,
    "comments": "Approved the document."
  }
}

The result creates another workflow instance. Is there something that I'm missing?

**Comment by glime-ai** [comment]
To assist you effectively, I need to clarify a few points:

1. When you say you invoke the workflow "POST" without a body or params, are you referring to initially starting the workflow instance, which then gets suspended waiting for an event?

2. The JSON payload you've posted for Postman, is this the payload you're using to trigger the "Approve" event on the suspended workflow instance?

3. Are you using the correct endpoint to trigger the event for the suspended workflow? Typically, triggering an event would be done through an endpoint specific to events, not by creating a new workflow instance.

4. Have you verified that the "eventName" you are using matches exactly with the event name expected by the "Event" activity in your workflow?

5. Could you provide the exact endpoint URL you are posting to in Postman?

Once you provide this information, I can give you a more accurate answer. If you're indeed using the correct endpoint and the event name matches, it's possible that there might be a configuration issue or a bug in the version of Elsa Studio you are using. If it's a bug, checking the Elsa GitHub issues page for similar reports or updating to the latest version might resolve the issue.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by doubleG7** [comment]
I figured it out, you need to pass workflowInstanceId as a parameter. 

https://workflow-engine-dev/http-enabled-workflows/listenandpublish?versionOptions=1&workflowInstanceId=1350376d1910d660
